hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneManager.java

```diff
@@ -1453,7 +1559,13 @@ public void updatePeerList(List<String> omNodeIds) {
       // Check if the OM NodeID is already present in the peer list or its
       // the local NodeID.
       if (!peerNodesMap.containsKey(omNodeId) && !isCurrentNode(omNodeId)) {
-        addOMNodeToPeers(omNodeId);
+        try {
+          addOMNodeToPeers(omNodeId);
+        } catch (IOException e) {
+          LOG.error("Fatal Error: Shutting down the system as otherwise it " +
```

Comment:
Question to understand: Why OM's will diverge? And also why are we shutting down OM here?

From PR description, i thought if config is not updated on existing OM's we are shutting down, but shutdown if new config does not have that node id is being done as part of this Jira.

Reply:
This function is called from the StateMachine. Any error in SM could be fatal as it diverges the OM states. Let's say  om1 and om2 update their configs about new nodes om4 and om5, but om3 does not as addOMNodeToPeers fails. This will lead to om3 being in a different config without having knowledge about om4 and om5. Now let's say om1 and om2 go down and om4 is the new leader. om3 cannot contact om4 to download a checkpoint if required as it does not know the addresses of om4.

Reply:
This function is called from OM SM. Any failure here could be fatal as OM states would diverge.
For example, let's say OM1 and OM2 successfully update their configs with newly bootstrapped nodes OM4 and OM5. OM3 fails to do so. Let's say later OM1 and OM2 shutdown with OM4 as the new leader. OM3 would not be able to download checkpoint from OM4 as it is not aware of OM4 addresses.